pkg/ingester/index/index.go

```diff
@@ -49,7 +49,6 @@ func (ii *InvertedIndex) Lookup(matchers []*labels.Matcher) []model.Fingerprint
 		result = append(result, fps...)
 	}
 
-	sort.Sort(fingerprints(result))
```

Comment:
"Remove the sort from the inverted index Lookup, its not actually needed."

I also add a comment in the place we call this discussion the locking.  Previously we used to ensure we locked the series in order for a query, to prevent deadlocks.  Now we only ever hold one series lock at once, so deadlocks can't happen.  Hence we don't need the sort.